pkgs/development/python-modules/redshift-connector/default.nix

```diff
@@ -0,0 +1,60 @@
+{ beautifulsoup4
+, boto3
+, buildPythonPackage
+, fetchFromGitHub
+, lib
+, lxml
+, packaging
+, pytest-mock
+, pytestCheckHook
+, pythonOlder
+, pytz
+, requests
+, scramp
+}:
+
+buildPythonPackage rec {
+  pname = "redshift-connector";
+  version = "2.0.910";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "amazon-redshift-python-driver";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-24yI6pXSHxhT30N3rJXAMtpCOhhGsBuDrwx9jMO1FW0=";
+  };
+
+  # disable test coverage
+  postPatch = ''
+    sed -i "/--cov/d" setup.cfg
+  '';
+
+  propagatedBuildInputs = [
+    beautifulsoup4
+    boto3
+    lxml
+    packaging
+    pytz
+    requests
+    scramp
+  ];
+
+  nativeCheckInputs = [
+    pytest-mock
+    pytestCheckHook
+  ];
+
+  # integration tests require a Redshift cluster
+  pytestFlagsArray = [ "test/unit" ];
```

Comment:
And finally the following to fix tests on darwin:

```suggestion
  pytestFlagsArray = [ "test/unit" ];
  
  __darwinAllowLocalNetworking = true;
```

````pytb
redshift-connector-aarch64-darwin> =================================== FAILURES ===================================
redshift-connector-aarch64-darwin> __________________ test_get_listen_socket_chooses_free_socket __________________
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin>     def test_get_listen_socket_chooses_free_socket():
redshift-connector-aarch64-darwin>         bacp: BrowserAzureCredentialsProvider = make_valid_browser_azure_credential_provider()
redshift-connector-aarch64-darwin>         ports: typing.Set[int] = set()
redshift-connector-aarch64-darwin>         sockets: typing.List["socket.socket"] = []
redshift-connector-aarch64-darwin>     
redshift-connector-aarch64-darwin>         for _ in range(5):  # open up 10 listen sockets
redshift-connector-aarch64-darwin> >           _socket: "socket.socket" = bacp.get_listen_socket()
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> test/unit/plugin/test_browser_azure_credentials_provider.py:129: 
redshift-connector-aarch64-darwin> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> self = <redshift_connector.plugin.browser_azure_credentials_provider.BrowserAzureCredentialsProvider object at 0x1038bb460>
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin>     def get_listen_socket(self: "BrowserAzureCredentialsProvider") -> socket.socket:
redshift-connector-aarch64-darwin>         s: socket.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
redshift-connector-aarch64-darwin> >       s.bind(("127.0.0.1", 0))  # bind to any free port
redshift-connector-aarch64-darwin> E       PermissionError: [Errno 1] Operation not permitted
redshift-connector-aarch64-darwin> 
redshift-connector-aarch64-darwin> redshift_connector/plugin/browser_azure_credentials_provider.py:37: PermissionError

Reply:
ahh thank you, that would've taken me a long time to figure out. Added.